### PR TITLE
Backend rewrite to Kotlin

### PR DESCRIPTION
**Pull request by falkecarlsen**
> What has been done?

Yes

- Rewrite the backend in Kotlin
- Add Gradle configuration
- Add sane .gitignore for project

### CODE REVIEW

File: src/test/kotlin/gwent/vanilla/AppTest.kt [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+import kotlin.test.Test
+import kotlin.test.assertTrue
+
+class AppTest {
+    @Test
+    fun dummyTest() {
+        assertTrue { true }
+    }
+}
```
Comment (NicEastvillage) [inline src/test/kotlin/gwent/vanilla/AppTest.kt:11]: Useless test

**Review by NicEastvillage — CHANGES_REQUESTED** [review]
One thing. The folder `src/public/` should be renamed to `src/resources`

File: src/test/kotlin/gwent/vanilla/AppTest.kt (hunk 1 above)
Comment (falkecarlsen) [reply src/test/kotlin/gwent/vanilla/AppTest.kt:11]: Yes, but just placeholder and useful for testing if JUnit and Gradle works - I vote it stays for the next five commits until proper tests are implemented.

**Comment by falkecarlsen** [comment]
> One thing. The folder `src/public/` should be renamed to `src/resources`

Fixed in 7661d04

**Pull request merged**